test/public.spec.js

```diff
@@ -0,0 +1,423 @@
+      beforeEach(function() {
+
+      });
+
+      it('should not show file size', function(done) {
+        createDomNodes([
+          createCrossDomainImg()
+        ]);
+        spyOn(window.performance, 'getEntriesByName').and.returnValue([{encodedBodySize: 0}]);
+        crossDomainImg.onload = function() {
+          window.NCC.imageChecker.showImagesInfo();
+
+          let textLines = getImageOverlayTextLines(0);
+          expect(textLines).toContain('File size unavailable');
+          done();
+        };
+      });
+    });
+  });
+
+  describe('hideImagesInfo()', function() {
+    it('should remove all image overlays', function() {
+      createDomNodes([
+        createBackgroundImg(),
+        createNoBackgroundImg(),
+        createHiddenImg(),
+        createInvisibleImg(),
+        createNoSrcImg(),
+        createNoDimensionImg(),
+        createSmallImg(),
+        createMediumImg(),
+        createBigImg(),
+        createImg()
+      ]);
+      window.NCC.imageChecker.showImagesInfo();
+
+      window.NCC.imageChecker.hideImagesInfo();
+
+      let siblings = getAllSiblingsAfter(images);
+      expect(siblings.length).toEqual(0);
+    });
+  });
+
+  describe('isImagesInfoActive()', function() {
+    it('should return true', function() {
+      createDomNodes([
+        createImg()
+      ]);
+      window.NCC.imageChecker.showImagesInfo();
+
+      expect(window.NCC.imageChecker.isImagesInfoActive()).toEqual(true);
+    });
+
+    it('should return false', function() {
+      createDomNodes();
+      window.NCC.imageChecker.showImagesInfo();
+
+      expect(window.NCC.imageChecker.isImagesInfoActive()).toEqual(false);
+    });
+  });
+
+  describe('_getTruncatedImageUrl()', function() {
+    it('should truncate url that does not fit bounding box', function() {
+      expect(window.NCC.imageChecker._getTruncatedImageUrl({
+        width: 50,
+        url: '1234567890'
+      })).toEqual('123...90');
+    });
+
+    it('should not truncate url that does fit bounding box', function() {
+      expect(window.NCC.imageChecker._getTruncatedImageUrl({
+        width: 50,
+        url: '12345678'
+      })).toEqual('12345678');
+    });
+  });
+
+  describe('_getElementTopLeft()', function() {
+    it('should use x y if available', function() {
+      createDomNodes([
+        createImg()
+      ]);
+      document.body.style = 'margin: 4px;';
+
+      expect(window.NCC.imageChecker._getElementTopLeft(img)).toEqual({top: 4, left: 4});
+    });
+
+    it('should use parent offset as fallback', function() {
+      createDomNodes([
+        createBackgroundImg()
+      ]);
+      document.body.style = 'margin: 2px;';
+
+      expect(window.NCC.imageChecker._getElementTopLeft(backgroundImg)).toEqual({top: 2, left: 2});
+    });
+  });
+
+  describe('_getImageCoverage()', function() {
+    it('should calculate using devicePixelRatio', function() {
+      expect(window.devicePixelRatio).toEqual(1);
```

Comment:
@tabanliviu BUG - This is at risk of failing if run using different plugins.  Please assign the value to `1` instead of asserting on it.